src/librbd/deep_copy/ObjectCopyRequest.cc

```diff
@@ -835,6 +861,34 @@ void ObjectCopyRequest<I>::compute_dst_object_may_exist() {
                    << dendl;
 }
 
+template <typename I>
+uint64_t ObjectCopyRequest<I>::compute_starting_end_size() {
+  uint64_t end_size = 0;
+
+  if (m_src_snap_id_start == 0) {
+    return 0;
+  }
+
+  for (auto& [key, image_intervals] : m_snapshot_delta) {
+    io::WriteReadSnapIds write_read_snap_ids{key};
+
+    // advance the src write snap id to the first valid snap id
+    if (write_read_snap_ids.first <= m_src_snap_id_start) {
```

Comment:
There can be multiple entries in the snapshot_delta if the mirror daemon has been down for a while and multiple mirror snapshots have been created in that interval.
Eg: 

```
2024-02-20T14:20:27.430+0530 7fe05e4586c0 20 librbd::deep_copy::ObjectCopyRequest: 0x564054b64480 ObjectCopyRequest: dst_oid=rbd_data.10a63265ae75.0000000000000000, src_snap_id_start=34, dst_snap_id_start=23, snap_map={35=[18446744073709551614,24,23]}
...
2024-02-20T14:20:27.431+0530 7fe05e4586c0 20 librbd::deep_copy::ObjectCopyRequest: 0x564054b64480 handle_list_snaps: snapshot_delta={34,34={0~4194304(4194304)},35,35={0~4194304(4194304)}}

```

Reply:
> and multiple mirror snapshots have been created in that interval.

There can be multiple snapshots created, but the range passed to list-snaps always starts with the first one.  Notice that in your example `src_snap_id_start=34` and that the first entry in snapshot delta is `34,34`.  I was asking whether you were expecting some other entries _before_ the entry corresponding to `m_src_snap_id_start` because that is what the loop kind of hints at.

Reply:
I think having a loop is safer as we may miss some scenario as I am not sure if there can be any scenario where there may be different entries.